src/lib/viewers/BaseViewer.js

```diff
@@ -493,6 +493,22 @@ class BaseViewer extends EventEmitter {
         });
     }
 
+    /**
+     * Emits a viewer metric
+     *
+     * @protected
+     * @emits metric
+     * @param {string} event - Event name
+     * @param {Object} data - Event data
+     * @return {void}
+     */
+    emitMetric(event, data) {
+        super.emit(VIEWER_EVENT.metric, {
+            event,
+            ...data
```

Comment:
I don't think we you want use spread syntax here. you want the `data` prop on the object emitted to equal `data`.